File: ruby/prime-factors/prime_factors.rb
require 'prime'

class PrimeFactors
  def self.for(num, factors = [], count = 2) 
    return factors if num < count
    if Prime.prime?(count) && num%count === 0
      factors << count
      remainder = num/count
      self.for(remainder, factors, count)
    else
      count += 1
      self.for(num, factors, count)
    end
    factors
  end
end

File: ecmascript/prime-factors/prime-factors.js
export default class PrimeFactors {
  for(num) {
    const arr = []
    let do_num = num
    let i = 2;
    do {
      if(do_num % i === 0){
        arr.push(i)
        do_num = do_num / i
      } else {
        i += 1
      }
    } while ( do_num > 1 )
  return arr
  }
}